assignment-client/src/messages/MessagesMixer.cpp

```diff
@@ -47,6 +47,13 @@ void MessagesMixer::handleMessages(QSharedPointer<ReceivedMessage> receivedMessa
     MessagesClient::decodeMessagesPacket(receivedMessage, channel, isText, message, data, senderID);
 
     auto nodeList = DependencyManager::get<NodeList>();
+    if (_allSubscribers.value(senderNode->getUUID()) >= _maxMessagesPerSecond) {
```

Comment:
hm...or does this handle when it’s not there?  maybe you should first check if it contains it?  can we use iterators to reduce the number of lookups we do here and below?